deal.ii build is broken with Clang 7.3.0

```
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/source/grid/grid_reordering.cc:736:11: error: call to 'cell_measure' is ambiguous
      if (GridTools::cell_measure<2>(all_vertices, vertices_lex) < 0)
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/include/deal.II/grid/grid_tools.h:166:10: note: candidate function [with dim = 2, spacedim = 2]
  double cell_measure (const std::vector<Point<spacedim> > &all_vertices,
         ^
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/include/deal.II/grid/grid_tools.h:1661:10: note: candidate function [with dim = 2]
  double cell_measure (const std::vector<Point<dim> > &all_vertices,
         ^
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/source/grid/grid_reordering.cc:749:23: error: call to 'cell_measure' is ambiguous
          AssertThrow(GridTools::cell_measure<2>(all_vertices, vertices_lex) > 0,
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/include/deal.II/base/exceptions.h:352:28: note: expanded from macro 'AssertThrow'
    if (__builtin_expect(!(cond), false))                                     \
                           ^~~~
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/include/deal.II/grid/grid_tools.h:166:10: note: candidate function [with dim = 2, spacedim = 2]
  double cell_measure (const std::vector<Point<spacedim> > &all_vertices,
         ^
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/include/deal.II/grid/grid_tools.h:1661:10: note: candidate function [with dim = 2]
  double cell_measure (const std::vector<Point<dim> > &all_vertices,
         ^
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/source/grid/grid_reordering.cc:1685:11: error: call to 'cell_measure' is ambiguous
      if (GridTools::cell_measure<3>(all_vertices, vertices_lex) < 0)
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/include/deal.II/grid/grid_tools.h:166:10: note: candidate function [with dim = 3, spacedim = 3]
  double cell_measure (const std::vector<Point<spacedim> > &all_vertices,
         ^
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/include/deal.II/grid/grid_tools.h:1661:10: note: candidate function [with dim = 3]
  double cell_measure (const std::vector<Point<dim> > &all_vertices,
         ^
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/source/grid/grid_reordering.cc:1700:23: error: call to 'cell_measure' is ambiguous
          AssertThrow(GridTools::cell_measure<3>(all_vertices, vertices_lex) > 0,
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/include/deal.II/base/exceptions.h:352:28:[ 46%] Building CXX object source/fe/CMakeFiles/obj_fe.release.dir/fe_dg_vector.cc.o
 note: expanded from macro 'AssertThrow'
    if (__builtin_expect(!(cond), false))                                     \
                           ^~~~
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/include/deal.II/grid/grid_tools.h:166:10: note: candidate function [with dim = 3, spacedim = 3]
  double cell_measure (const std::vector<Point<spacedim> > &all_vertices,
         ^
/private/var/folders/5k/sqpp24tx3ylds4fgm13pfht00000gn/T/davydden/spack-stage/spack-stage-VJkWhp/dealii/include/deal.II/grid/grid_tools.h:1661:10: note: candidate function [with dim = 3]
  double cell_measure (const std::vector<Point<dim> > &all_vertices,
         ^
```

@drwells could you have a look? I think it is related to https://github.com/dealii/dealii/commit/18ddfc4cf18c91d12c3af7dd9c71342342b26eaf

--- a/include/deal.II/grid/grid_tools.h
+++ b/include/deal.II/grid/grid_tools.h
@@ -157,30 +157,18 @@ namespace GridTools
    * get the same result using <code>cell-@>measure()</code>, but this
    * function also works for cells that do not exist except that you make it
    * up by naming its vertices from the list.
-   *
-   * @note Since <code>dim</code> does not explicitly appear in the function
-   * signature, you must explicitly specify both dimensions when using this
-   * function.
-   */
-  template <int dim, int spacedim>
-  double cell_measure (const std::vector<Point<spacedim> > &all_vertices,
-                       const unsigned int (&vertex_indices)[GeometryInfo<dim>::vertices_per_cell]);
-
-
-  /**
-   * Same as the last function, but for the codimension zero case. This extra
-   * function allows one to write
-   *
-   * @code
-   * cell_measure(vertices, indices);
-   * @endcode
-   *
-   * and have the template function argument (i.e. <code>dim</code>) inferred
-   * automatically.
    */
   template <int dim>
   double cell_measure (const std::vector<Point<dim> > &all_vertices,
                        const unsigned int (&vertex_indices)[GeometryInfo<dim>::vertices_per_cell]);
+
+  /**
+   * A version of the last function that can accept input for nonzero
+   * codimension cases. This function only exists to aid generic programming
+   * and calling it will just raise an exception.
+   */
+  template <int dim, typename T>
+  double cell_measure (const T &, ...);
 
   /*@}*/
   /**
@@ -1655,16 +1643,12 @@ namespace GridTools
 
 namespace GridTools
 {
-  // This function just wraps the general case: see the note in its
-  // documentation above.
-  template <int dim>
-  double cell_measure (const std::vector<Point<dim> > &all_vertices,
-                       const unsigned int (&vertex_indices)[GeometryInfo<dim>::vertices_per_cell])
+  template <int dim, typename T>
+  double cell_measure (const T &, ...)
   {
-    return cell_measure<dim, dim>(all_vertices, vertex_indices);
+    Assert(false, ExcNotImplemented());
+    return std::numeric_limits<double>::quiet_NaN();
   }
-
-
 
   template <int dim, typename Predicate, int spacedim>
   void transform (const Predicate    &predicate,

--- a/source/grid/grid_tools.cc
+++ b/source/grid/grid_tools.cc
@@ -170,7 +170,7 @@ namespace GridTools
 
   template <>
   double
-  cell_measure<1, 1>
+  cell_measure<1>
   (const std::vector<Point<1> > &all_vertices,
    const unsigned int (&vertex_indices)[GeometryInfo<1>::vertices_per_cell])
   {
@@ -182,43 +182,7 @@ namespace GridTools
 
   template <>
   double
-  cell_measure<1, 2>
-  (const std::vector<Point<2> > &,
-   const unsigned int (&)[GeometryInfo<1>::vertices_per_cell])
-  {
-    Assert(false, ExcNotImplemented());
-    return std::numeric_limits<double>::quiet_NaN();
-  }
-
-
-
-  template <>
-  double
-  cell_measure<1, 3>
-  (const std::vector<Point<3> > &,
-   const unsigned int (&)[GeometryInfo<1>::vertices_per_cell])
-  {
-    Assert(false, ExcNotImplemented());
-    return std::numeric_limits<double>::quiet_NaN();
-  }
-
-
-
-  template <>
-  double
-  cell_measure<2, 3>
-  (const std::vector<Point<3> > &,
-   const unsigned int (&)[GeometryInfo<2>::vertices_per_cell])
-  {
-    Assert(false, ExcNotImplemented());
-    return std::numeric_limits<double>::quiet_NaN();
-  }
-
-
-
-  template <>
-  double
-  cell_measure<3, 3>
+  cell_measure<3>
   (const std::vector<Point<3> > &all_vertices,
    const unsigned int (&vertex_indices)[GeometryInfo<3>::vertices_per_cell])
   {
@@ -342,7 +306,7 @@ namespace GridTools
 
   template <>
   double
-  cell_measure<2, 2>
+  cell_measure<2>
   (const std::vector<Point<2> > &all_vertices,
    const unsigned int (&vertex_indices) [GeometryInfo<2>::vertices_per_cell])
   {

--- a/source/grid/tria.cc
+++ b/source/grid/tria.cc
@@ -1652,7 +1652,7 @@ namespace internal
                 // throw an exception if no such cells should exist.
                 if (!triangulation.check_for_distorted_cells)
                   {
-                    const double cell_measure = GridTools::cell_measure<1, spacedim>
+                    const double cell_measure = GridTools::cell_measure<1>
                                                 (triangulation.vertices, cells[cell_no].vertices);
                     AssertThrow(cell_measure > 0, ExcGridHasInvalidCell(cell_no));
                   }
@@ -1851,7 +1851,7 @@ namespace internal
                 // See the note in the 1D function on this if statement.
                 if (!triangulation.check_for_distorted_cells)
                   {
-                    const double cell_measure = GridTools::cell_measure<2, spacedim>
+                    const double cell_measure = GridTools::cell_measure<2>
                                                 (triangulation.vertices, cells[cell_no].vertices);
                     AssertThrow(cell_measure > 0, ExcGridHasInvalidCell(cell_no));
                   }
@@ -2234,7 +2234,7 @@ namespace internal
             // See the note in the 1D function on this if statement.
             if (!triangulation.check_for_distorted_cells)
               {
-                const double cell_measure = GridTools::cell_measure<3, spacedim>
+                const double cell_measure = GridTools::cell_measure<3>
                                             (triangulation.vertices, cells[cell_no].vertices);
                 AssertThrow(cell_measure > 0, ExcGridHasInvalidCell(cell_no));
               }

